pkg/kademlia/routing_integration_test.go

```diff
@@ -0,0 +1,458 @@
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	table, close := routingCtor(PadID("a3", "3"), 5, 2, 0)
+	defer close()
+
+	for _, prefix2 := range "08" {
+		for _, prefix1 := range "b4f25c896de03a71" {
+			require.NoError(t, table.ConnectionSuccess(
+				NodeFromPrefix(string([]rune{prefix1, prefix2}), "f")))
+		}
+	}
+
+	nodes, err := table.FindNear(PadID("c7139", "1"), 1)
+	require.NoError(t, err)
+	requireNodesEqual(t, []*pb.Node{
+		NodeFromPrefix("c0", "f"),
+	}, nodes)
+
+	require.NoError(t, table.ConnectionSuccess(
+		Node(PadID("c0", "f"), "new-address:3")))
+
+	nodes, err = table.FindNear(PadID("c7139", "1"), 1)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(nodes))
+	require.Equal(t, PadID("c0", "f"), nodes[0].Id)
+	require.Equal(t, "new-address:3", nodes[0].Address.Address)
+}
+
+func TestUpdateCache_Routing(t *testing.T)     { testUpdateCache(t, newRouting) }
+func TestUpdateCache_TestRouting(t *testing.T) { testUpdateCache(t, newTestRouting) }
+func testUpdateCache(t *testing.T, routingCtor routingCtor) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	table, close := routingCtor(PadID("a3", "3"), 1, 1, 0)
+	defer close()
+
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("80", "0")))
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("c0", "0")))
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("40", "0")))
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("00", "0")))
+
+	require.NoError(t, table.ConnectionSuccess(Node(PadID("00", "0"), "new-address:6")))
+	require.NoError(t, table.ConnectionFailed(NodeFromPrefix("40", "0")))
+
+	nodes, err := table.FindNear(PadID("00", "0"), 4)
+	require.NoError(t, err)
+
+	requireNodesEqual(t, []*pb.Node{
+		Node(PadID("00", "0"), "new-address:6"),
+		NodeFromPrefix("80", "0"),
+		NodeFromPrefix("c0", "0"),
+	}, nodes)
+}
+
+func TestFailureUnknownAddress_Routing(t *testing.T)     { testFailureUnknownAddress(t, newRouting) }
+func TestFailureUnknownAddress_TestRouting(t *testing.T) { testFailureUnknownAddress(t, newTestRouting) }
+func testFailureUnknownAddress(t *testing.T, routingCtor routingCtor) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	table, close := routingCtor(PadID("a3", "3"), 1, 1, 0)
+	defer close()
+
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("80", "0")))
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("c0", "0")))
+	require.NoError(t, table.ConnectionSuccess(Node(PadID("40", "0"), "address:2")))
+	require.NoError(t, table.ConnectionSuccess(NodeFromPrefix("00", "0")))
+	require.NoError(t, table.ConnectionFailed(NodeFromPrefix("40", "0")))
+
+	nodes, err := table.FindNear(PadID("00", "0"), 4)
+	require.NoError(t, err)
+
+	requireNodesEqual(t, []*pb.Node{
+		Node(PadID("40", "0"), "address:2"),
+		NodeFromPrefix("80", "0"),
+		NodeFromPrefix("c0", "0"),
+	}, nodes)
+}
+
+func TestShrink_Routing(t *testing.T)     { testShrink(t, newRouting) }
```

Comment:
Error Trace:	routing_integration_test.go:435
        	            				routing_integration_test.go:415
        	Error:      	Received unexpected error:
        	            	routing table error: could not remove node routing table error: could not get k bucket routing table error: could not find k bucket
        	            		storj.io/storj/pkg/kademlia.(*RoutingTable).ConnectionFailed:211
        	            		storj.io/storj/pkg/kademlia.testShrink:435
        	            		storj.io/storj/pkg/kademlia.TestShrink_Routing:415
        	            		testing.tRunner:827
        	Test:       	TestShrink_Routing

Reply:
I think this error stems from the fact that we can't find the k bucket from the ID of all 0's.

Reply:
okay latest changes no longer use all zeros